src/app/donation-start/donation-start.component.ts

```diff
@@ -750,6 +770,32 @@ export class DonationStartComponent implements AfterContentChecked, AfterContent
     }
   }
 
+  onBillingPostCodeChanged(event: Event) {
+    // If previous payment attempt failed due to incorrect post code
+    // and the post code has just been changed again, clear stripeError
+    // and clear stripeResponseErrorCode. This is because if we don't,
+    // then when the user goes back to the Payment details step and
+    // updates their post code, pressing the 'Next' button will keep them
+    // where they are and not proceed them to the next step, because the
+    // next() method calls goToFirstVisibleError().
+    if (this.isBillingPostCodeInvalid()) {
+      this.stripeError = undefined;
+      this.stripeResponseErrorCode = undefined;
+
+      // Remove the ValidateBillingPostCode custom validator so billing postcode
+      // doesn't show as invalid after a change
+      this.paymentGroup.controls.billingPostcode.setValidators([
```

Comment:
@alihejazi  could call `addStripeCardBillingValidators()` here so we're not maintaining it in more places than needed? I believe we're already relying on this to clear your extra validator on card # changes so this also makes the implementation self-consistent for both types of correction.